### PR TITLE
[Frontend_V2] Limited access to unverified users

### PR DESCRIPTION
**Pull request by Kajol-Kumari**
Fixes: #2646 [Redirecting to home page with limited access after sign up]

**Changes introduced through this PR**

- [x] Permission denied component Added
 

- [x] Unverified users can view `all challenges`

- [x] Unverified users won't be able to see `dashboard`, `hosted challenges`, `participant teams` and `hosted teams`.

- [x] Users will be directed to home page instead of login page after logging out.

GIF of feature:
ezgif com-video-to-gif

cc: @deshraj @Sanji515 @RishabhJain2018 @lunayach @Shekharrajak

### CODE REVIEW

File: frontend_v2/src/app/services/auth.service.ts [hunk 1]
```diff
@@ -69,7 +68,7 @@ export class AuthService {
      */
     logOut() {
       const temp = {isLoggedIn: false};
-      const routePath = '/auth/login';
+      const routePath = '';
```
Comment (Sanji515) [inline frontend_v2/src/app/services/auth.service.ts:71]: @Kajol-Kumari Why we are redirecting it to landing page after logout?

**Comment by Sanji515** [comment]
@Kajol-Kumari Also the UI of sidebar doesn't look same as in the master branch so is this the expected change of this PR? If not then please fix it.

File: frontend_v2/src/app/services/auth.service.ts (hunk 1 above)
Comment (Kajol-Kumari) [reply frontend_v2/src/app/services/auth.service.ts:71]: Hey @Sanji515 I have done this change as per the discussion done in [here](https://github.com/Cloud-CV/EvalAI/issues/2646) 
If it seems to be not needed currently, please let me know. I'll remove it.

**Comment by Kajol-Kumari** [comment]
> @Kajol-Kumari Also the UI of sidebar doesn't look same as in the master branch so is this the expected change of this PR? If not then please fix it.

Hey @Sanji515 I haven't done any change in the sidebar. The `Dashboard` has been removed from the side-bar through [this](https://github.com/Cloud-CV/EvalAI/pull/2851)PR.Can you please just point to the change except the dashboard removal.

**Comment by RishabhJain2018** [comment]
> you can take a look here if we want to redirect to the landing page or not after logout.
@Kajol-Kumari let's redirect them to all challenges page.

**Pull request merged**